### ISSUE TITLE
Make PublicEvent(); default to PublicEvent<undefined>();

### ISSUE THREAD

**Issue by soimon** [JacobFischer/ts-typed-events#16]
The release candidate is looking really good so far! I have noted a small inconsistency in the construction of `PublicEvent` that seems unnecessary and makes gradual adoption from 2.0 to 3.0 a little harder. It has to do with the creation of public signals:
```typescript
const event = new PublicEvent();
event.emit(); // Not permitted as it expects a parameter with type unknown

const event = createPublicEventEmitter();
event.emit(); // All good
```
I'm in between deadlines at the moment, otherwise I would've made a PR. I might get back at it later.
Thanks for the great work so far!

**Comment by JacobFischer** [comment]
The idea behind `createPublicEventEmitter` it to mirror `createEventEmitter`'s API. Both return an emitter function for their respect event types.

If you want to create the `PublicEvent` directly, you can do that (essentially just like in < v3):
```js
const event = new PublicEvent();
event.emit();
```

_However_, when I tested that code in the latest V3 release, it indeed gives the following error for `emit()`:
```
Expected 1 arguments, but got 0.ts(2554)
An argument for 'emitting' was not provided.
```

**This is a bug.** The intention for this entire module is that if you omit a type, then you are not emitting anything other than the triggering of the event itself. Therefore `emitting` arguments should be omitted.

I will work on fixing this bug today and releasing another V3 release candidate with this fix.

Once that bug is fixed, does the intended usage of `PublicEvent` meet your requirements?

**Comment by JacobFischer** [comment]
The default is fixed via 30d7e5493d1823318682808124bf23c1b65e4ee9, and has been released in `v3.0.0-rc.4`.

If this copy of the code looks good I'll be intending to release the true v3 release by the end of the month.

**Comment by soimon** [comment]
> Once that bug is fixed, does the intended usage of `PublicEvent` meet your requirements?

It does indeed, as `PublicEvent` is now backwards compatible with `Event` from v2.
This allows for gradual adoption by changing the existing imports to
```typescript
import {PublicEvent as Event} from 'ts-typed-events';
```
(not necessarily a smart thing to do, but being able to is nice while prototyping).
